From bb66c95f43e2cb727a9aafbce8c70f25df513a13 Mon Sep 17 00:00:00 2001
Message-Id: <bb66c95f43e2cb727a9aafbce8c70f25df513a13.1575896661.git.liu.denton@gmail.com>
In-Reply-To: <cover.1575896661.git.liu.denton@gmail.com>
References: <cover.1575896661.git.liu.denton@gmail.com>
From: Denton Liu <liu.denton@gmail.com>
Date: Mon, 9 Dec 2019 03:19:06 -0800
Subject: [PATCH 2/5] notes: create init_display_notes() helper
To: Git Mailing List <git@vger.kernel.org>
Cc: Elijah Newren <newren@gmail.com>,
    Junio C Hamano <gitster@pobox.com>,
    Beat Bolli <dev+git@drbeat.li>,
    Pavel Roskin <plroskin@gmail.com>

We currently open code the initialization for revs->notes_opt. Abstract
this away into a helper function so that the logic can be reused in a
future commit.

This is slightly wasteful as we memset the struct twice but this is only
run once so it shouldn't have any major effect.

Signed-off-by: Denton Liu <liu.denton@gmail.com>
---
 notes.c    | 6 ++++++
 notes.h    | 5 +++++
 revision.c | 2 +-
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/notes.c b/notes.c
index fd6cef14a3..53d1e7767c 100644
--- a/notes.c
+++ b/notes.c
@@ -1039,6 +1039,12 @@ struct notes_tree **load_notes_trees(struct string_list *refs, int flags)
 	return trees;
 }
 
+void init_display_notes(struct display_notes_opt *opt)
+{
+	memset(opt, 0, sizeof(*opt));
+	opt->use_default_notes = -1;
+}
+
 void load_display_notes(struct display_notes_opt *opt)
 {
 	char *display_ref_env;
diff --git a/notes.h b/notes.h
index 1ce528442a..c0b712371c 100644
--- a/notes.h
+++ b/notes.h
@@ -260,6 +260,11 @@ struct display_notes_opt {
 	struct string_list extra_notes_refs;
 };
 
+/*
+ * Initialize a display_notes_opt to its default value.
+ */
+void init_display_notes(struct display_notes_opt *opt);
+
 /*
  * Load the notes machinery for displaying several notes trees.
  *
diff --git a/revision.c b/revision.c
index d4aaf0ef25..24ad974590 100644
--- a/revision.c
+++ b/revision.c
@@ -1637,7 +1637,7 @@ void repo_init_revisions(struct repository *r,
 		revs->diffopt.prefix_length = strlen(prefix);
 	}
 
-	revs->notes_opt.use_default_notes = -1;
+	init_display_notes(&revs->notes_opt);
 }
 
 static void add_pending_commit_list(struct rev_info *revs,
-- 
2.24.0.627.geba02921db

